src/main/Extensions/Base64Conversion/Base64Conversion.ts

```diff
@@ -0,0 +1,87 @@
+import { AssetPathResolver } from "@Core/AssetPathResolver";
+import { Extension } from "@Core/Extension";
+import { Translator } from "@Core/Translator";
+import { SearchResultItem, SearchResultItemActionUtility } from "@common/Core";
+import { Image } from "@common/Core/Image";
+import { Buffer } from "buffer";
+import { InvocationArgument } from "./InvocationArgument";
+
+export class Base64Conversion implements Extension {
+    public readonly id = "Base64Conversion";
+    public readonly name = "Base64 Conversion";
+
+    public readonly author = {
+        name: "Christopher Steiner",
+        githubUserName: "ChristopherSteiner",
+    };
+
+    private readonly defaultSettings = {};
+
+    public constructor(
+        private readonly assetPathResolver: AssetPathResolver,
+        private readonly translator: Translator,
+    ) {}
+
+    public async getSearchResultItems(): Promise<SearchResultItem[]> {
+        const { t } = this.translator.createT(this.getI18nResources());
+
+        return [
+            {
+                id: "Base64Converison:invoke",
+                description: t("searchResultItemDescription"),
+                name: t("searchResultItemName"),
+                image: this.getImage(),
+                defaultAction: SearchResultItemActionUtility.createInvokeExtensionAction({
+                    extensionId: this.id,
+                    description: t("searchResultItemActionDescription"),
+                    fluentIcon: "OpenRegular",
+                }),
+            },
+        ];
+    }
+
+    public async invoke(argument: InvocationArgument): Promise<string> {
+        if (argument.action === "encode") {
+            return Buffer.from(argument.payload, "binary").toString("base64");
+        } else {
+            return Buffer.from(argument.payload, "base64").toString("binary");
+        }
+    }
```

Comment:
This is just my personal preference, but I like to do the following:

- I like to destructure objects in function/method arguments
- If I encounter two return statements in an `if/else` then I like to turn it to a single return statement

Here is an example:

```typescript
public async invoke({ action, payload }: InvocationArgument): Promise<string> {
    return action === "encode"
        ? Buffer.from(payload, "binary").toString("base64")
        : Buffer.from(payload, "base64").toString("binary");
}
```

In my opinion the code looks more pleasing and elegant to the eye and is therefore nicer to read.